src/main/java/io/github/syst3ms/skriptparser/lang/entries/SectionConfiguration.java

```diff
@@ -0,0 +1,125 @@
+package io.github.syst3ms.skriptparser.lang.entries;
+
+import io.github.syst3ms.skriptparser.file.FileSection;
+import io.github.syst3ms.skriptparser.file.VoidElement;
+import io.github.syst3ms.skriptparser.lang.CodeSection;
+import io.github.syst3ms.skriptparser.log.ErrorType;
+import io.github.syst3ms.skriptparser.log.SkriptLogger;
+import io.github.syst3ms.skriptparser.parsing.ParserState;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SectionConfiguration {
+	@Nullable
+	private CodeSection parent;
+	private final List<EntryLoader> entries = new ArrayList<>();
+	private final Map<String, Object> data = new HashMap<>();
+
+	public SectionConfiguration addOption(String key) {
+		entries.add(new OptionLoader(false, key, false));
+		return this;
+	}
+
```

Comment:
Optionals will be used frequently, so I think it'll be best to have a method for adding optionals.
```suggestion

    /**
     * Adds an optional entry to this {@link SectionConfiguration}
     * 
     * @param key The key of the entry.
     * @return This {@link SectionConfiguration} with the newly added entry for building.
     */
    public SectionConfiguration addOptional(String key) {
        entries.add(new OptionLoader(false, key, true));
        return this;
    }

```

Reply:
I think I'm going to refactor the 'optional' option resulting in easier registration, because the current system is indeed a bit cumbersome.